### DISCUSSION TITLE
Oceanic moho inversion.

### DISCUSSION THREAD

**Discussion by geojoeyphillips** [mdtanker/invert4geom#56] [Q&A] [answered]
Hi Matt, this package looks really great and exactly what I have been looking for after weeks of trying other basement inversion codes. Your thesis and codes are very readable and make sense after I've spent some time looking at the fatiando Harmonica gravity codes. The question I have is tying together the real data from [Ross Ice Shelf Inversion](https://github.com/mdtanker/RIS_gravity_inversion/blob/main/notebooks/Ross_Ice_Shelf_inversion/RIS_inversion.ipynb) to the example [combining it all together codes.](https://github.com/mdtanker/invert4geom/blob/main/docs/user_guide/combining_it_all.ipynb) Pardon my ignorance but I am struggling to see what to replace synthetic topography with especially with a surface we have a rough understanding from surface waves and forward gravity modeling (Moho). I have the bathymetry and residual Bouguer disturbance calculated but if you were to use these codes to understand the oceanic Moho, what would your starting model be? A flat plane ~6km below? A density contrast of 2800 oceanic crust to 3300 mantle makes sense (neglecting a thermal anomaly calculation for now) but the zref should then be 6 km below the bathymetry? I would greatly appreciate your input in relating these notebooks together with a new dataset. Thank you!

**Comment by mdtanker** [comment, marked as answer]
Hi @geojoeyphillips thanks for opening a discussion. I'm glad to see someone has had a look at the package! Let me know if I'm misunderstanding your questions, but your starting model should be an estimation of the layer of interest, in your case the Moho. 

If you know nothing about the Moho topography, then yes the starting layer would be a flat surface at a reasonable depth. If you had a few points where you know the Moho depth, say from a seismic survey, then you should build a starting layer with these points. For this, you could interpolate between the known points with something like [Verde](https://www.fatiando.org/verde/latest/tutorials/overview.html)). 

With this starting layer, you can then make a starting model (layer of prisms). For this you'll need to choose an appropriate density contrast as you said, and a reference level (zref). I think a reasonable starting points for zref would be the mean elevation of the starting layer.  

If you did have any constraints on Moho depth, you can use these constraint points in a cross-validation scheme to choose the optimal values for density contrast and zref, as shown in the [density CV example](https://invert4geom.readthedocs.io/en/latest/user_guide/density_cross_validation.html) and the [zref CV example](https://invert4geom.readthedocs.io/en/latest/user_guide/reference_level_cross_validation.html).

As I have only used this inversion so far for modelling bathymetry, I actually haven't tried it for modelling the Moho yet, so I'll be very interested to see how it performs for you. Lots of this package is based on [Uieda and Barbosa 2016](https://academic.oup.com/gji/article-lookup/doi/10.1093/gji/ggw390) who used a very similar inversion to model the Moho below South America. You can see from Figure 7 that through the cross-validation routines, they settled on a density contrast of 350 kg m−3 and a zref of -30km. They didn't incorporate the constraints in their starting inversion and started with a flat Moho at -60km. 

Please let me know how this goes, as there may need to be changes to the code to make it work better for the application to model the Moho.

**Reply by geojoeyphillips** [reply]
Hey @mdtanker thanks for the quick reply! This all sounds great and this package will be super helpful for a couple projects so I'm excited to dig deeper. Thank you for the clarification about zref, I believe the mistake I made was using the bathymetry grid as the starting_prism and setting zref = -20 km so it made a ~25 km thick layer. These are below sea level correct? For ~5km water depth and ~6 km thick crust then -20 km will be enough for thinner oceanic crust. 

For the gravity observations the example sets 'upward' to 1000, this is the equivalent of having all observations at 1000 m elevation? In the oceans the observation height would be 0 or the height of the geoid at that grid point, but we're effectively downward continuing with the Bouguer disturbance so the 'upward' in this case would be the seafloor elevation (-5000m)? This is really specific to the oceans but I think once I understand these concepts I should have an initial inversion working with a plausible result. Thank you so much!

**Reply by mdtanker** [reply]
Sorry I missed this reply. Yeah all the elevation values are negative downwards, and yes upwards refers to the elevation of the gravity observations. How was your gravity data collected? Was it derivrd from satellite altimetry? If so, that is a good question for what the observation height should be, not too sure about that. Maybe look at what Uieda did in the paper I linked above. 

Let me know how the inversion goes. I'll try and find a good test dataset for a moho inversion and include it in a new tutorial.
